### DISCUSSION TITLE
Capture exit code from os.exec(v)

### DISCUSSION THREAD

**Discussion by dkaszews** [xmake-io/xmake#5085] [Q&A] [answered]
I want to use phony targets as transparent helper scripts for running build results. However, if any of `os.exec(v)` return a non-zero exit code, let's say `1`, xmake aborts execution, prints `error: execv(...) failed(1)` and returns `255`. This messes with my test setup. I would prefer xmake to just exit with status `1` and no extra output.

Is there any easy way to capture the exit code? I tried `pcall` but it does not exist. I then found the [`try-catch`](https://xmake.io/mirror/manual/builtin_modules.html#try-catch-finally) construct, but `errors` only contains the `error: execv(...) failed(1)` line which I would rather not parse as it is does not seem a stable solution.

Something like `stdout, stderr, result = os.iorun(...)` would be perfect.

**Comment by waruqi** [comment, marked as answer]
`os.execv(program, argv, {try = true, stdout = os.nuldev(), stderr = os.nuldev()})`

**Reply by dkaszews** [reply]
Thank you, that is perfect! Are all the options of `os.exec(v)` documented somewhere? Searching for `try = true` I only see it on `import`.

**Reply by waruqi** [reply]
https://xmake.io/#/manual/builtin_modules?id=osexecv

**Reply by dkaszews** [reply]
Yes, but it does not mention the `try` option.

**Reply by dkaszews** [reply]
Also, this option seems to be ignored by `os.iorunv`, so if I need to capture `stdout` and/or `stderr`, I need to do it via temporary files.

**Reply by waruqi** [reply]
only os.execv supports it, or you can use `try-catch`
